src/dataframe/insertion.jl

```diff
@@ -452,18 +526,38 @@ julia> push!(df, NamedTuple(), cols=:subset)
    6 │ 11            12  missing
    7 │ 1.0      missing        1.0
    8 │ missing  missing  missing
+
+julia> push!(DataFrame(a=1, b=2), (3, 4), (b=6, a=5))
+3×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1      2
+   2 │     3      4
+   3 │     5      6
 ```
 """
-Base.push!(df::DataFrame, row::Any; promote::Bool=false) =
-    _row_inserter!(df, -1, row, Val{:push}(), promote)
+function Base.push!(df::DataFrame, row::Any;
+                    cols=:setequal, promote::Bool=false)
+    if !(cols in (:setequal, :orderequal))
+        throw(ArgumentError("`cols` keyword argument must be " *
+                            ":orderequal or :setequal"))
+    end
+
+    return _row_inserter!(df, -1, row, Val{:push}(), promote)
+end
 
 """
-    pushfirst!(df::DataFrame, row::Union{Tuple, AbstractArray}; promote::Bool=false)
+    pushfirst!(df::DataFrame, row::Union{Tuple, AbstractArray};
+               cols::Symbol=:setequal, promote::Bool=false)
     pushfirst!(df::DataFrame, row::Union{DataFrameRow, NamedTuple, AbstractDict,
                                          Tables.AbstractRow};
                cols::Symbol=:setequal, promote::Bool=(cols in [:union, :subset]))
+    pushfirst!(df::DataFrame, rows...;
```

Comment:
Note that I have reverted this part of your update of the docs. The reason is that `rows...` allows for mixing of the first and second style (named and unnamed rows) in the current design. I could disallow this if you prefer, see the comment https://github.com/JuliaData/DataFrames.jl/pull/3372/files#r1314234174